src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -381,47 +394,132 @@ private void DropItems(Character killer)
+						maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
+					}
+
+					dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+
+					if (killer == null || dropChance > autoloot)
+					{
+						var direction = new Direction(rnd.Next(0, 360));
+						var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
+						var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+
+						dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
+						dropItem.Drop(this.Map, this.Position, direction, distance);
+					}
+					else
+					{
+						killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
+					}
 				}
+			}
+		}
 
-				var dropItem = new Item(itemData.Id);
-				var minAmount = dropItemData.MinAmount;
-				var maxAmount = dropItemData.MaxAmount;
+		/// <summary>
+		/// Rolls for a chance for this monster to become a rare (jackpot or elite) monster
+		/// </summary>
+		/// <param name="redOrb">true if this was spawned by a red orb and should get the boosted rates</param>
+		/// <returns></returns>
+		public void PossiblyBecomeRare(bool redOrb = false)
+		{
+			var rnd = RandomProvider.Get();
 
-				if (dropItemData.ItemId == ItemId.Silver || dropItemData.ItemId == ItemId.Gold)
-				{
-					minAmount = Math.Max(1, (int)(minAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-					maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-				}
+			var worldConf = ZoneServer.Instance.Conf.World;
 
-				dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+			var spawnRateBoost = 1F;
+			if (redOrb)
+				spawnRateBoost = worldConf.RedOrbJackpotRate / 100F;
 
-				if (killer == null || dropChance > autoloot)
-				{
-					var direction = new Direction(rnd.Next(0, 360));
-					var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
-					var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+			if (rnd.NextDouble() * 100D < worldConf.GoldJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.TwinkleBuff, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
 
-					dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
-					dropItem.Drop(this.Map, this.Position, direction, distance);
-				}
-				else
-				{
-					killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
-				}
+			if (rnd.NextDouble() * 100D < worldConf.BlueJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperExp, 1, 0, TimeSpan.Zero, this);
+				return;
 			}
+
+			if (rnd.NextDouble() * 100D < worldConf.SilverJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperDrop, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
+
+			// Check to see if this monster can become elite
+			// Note that on official it uses the map level rather than the monster level
```

Comment:
Why not implement the official way then/make it configurable?

Reply:
My main reason for thinking to do it this way is simply because it makes more sense.  If you take a red orb and summon a monster from it, it makes little sense that whether or not it could become elite depends on the map where you summon it.  I also did consider having a customization option for it, but it only really matters if EliteMinLevel is used, so we're basically setting two parameters that only matter if both are used.  I suppose we can just use the map level since if you set the EliteMinLevel to 0 (which I imagine would be the only other setting anyone would use) it does the same thing.

Reply:
> it makes more sense

Well, I get that, but do keep in mind that we're developing an _emulator_^^ The default behavior should always be the official behavior if possible.

Reply:
I noticed while trying to make it use the map that I don't think we actually currently have the maplevel, it's not even in maps.txt.